wrapper/src/main/java/software/amazon/jdbc/plugin/cache/DataRemoteCachePlugin.java

```diff
@@ -140,33 +147,76 @@ private ResultSet cacheResultSet(String queryStr, ResultSet rs, int expiry) thro
 
   /**
    * Determine the TTL based on an input query
-   * @param queryHint string. e.g. "NO CACHE", or "cacheTTL=100s"
+   * @param queryHint string. e.g. "CACHE_PARAM(ttl=300s)" or "CACHE_PARAM(ttl=100s, key=custom)"
    * @return TTL in seconds to cache the query.
    *         null if the query is not cacheable.
    */
   protected Integer getTtlForQuery(String queryHint) {
     // Empty query is not cacheable
     if (StringUtils.isNullOrEmpty(queryHint)) return null;
-    // Query longer than 16K is not cacheable
-    String[] tokens = queryHint.toLowerCase().split("cache");
-    if (tokens.length >= 2) {
-      // Handle "no cache".
-      if (!StringUtils.isNullOrEmpty(tokens[0]) && "no".equals(tokens[0])) return null;
-      // Handle "cacheTTL=Xs"
-      if (!StringUtils.isNullOrEmpty(tokens[1]) && tokens[1].startsWith("ttl=")) {
-        int endIndex = tokens[1].indexOf('s');
-        if (endIndex > 0) {
+    LOGGER.finest("queryHint is: " + queryHint);
+    // Find CACHE_PARAM anywhere in the hint string (case insensitive)
+    String upperHint = queryHint.toUpperCase();
+    int cacheParamStart = upperHint.indexOf("CACHE_PARAM(");
+    if (cacheParamStart == -1) return null;
+
+    // Find the matching closing parenthesis
+    int paramsStart = cacheParamStart + 12; // Length of "CACHE_PARAM("
+    int paramsEnd = queryHint.indexOf(")", paramsStart);
+    if (paramsEnd == -1) return null;
+
+    // Extract parameters between parentheses
+    String cacheParams = queryHint.substring(paramsStart, paramsEnd).trim();
+    LOGGER.finest("Cache params: " + cacheParams);
+
+    // Empty parameters
+    if (StringUtils.isNullOrEmpty(cacheParams)) {
+      LOGGER.warning("Empty CACHE_PARAM parameters");
```

Comment:
No need to log warning here. It is normal if the query hint doesn't contain cache parameters